### ISSUE TITLE
Status hooks of the long run job will be lost if jobservice is stopped

### ISSUE THREAD

**Issue by steven-zou** [goharbor/harbor#10679]
From the following log,

```log
Feb  7 08:09:44 172.18.0.1 jobservice[2587]: 2020-02-07T08:09:44Z [INFO] [/jobservice/runtime/bootstrap.go:195]: API server is serving at 8080 with [http] mode at node [244eca6dae82:172.18.0.9]

Feb  7 08:30:40 172.18.0.1 jobservice[2587]: 2020-02-07T08:30:40Z [INFO] [/jobservice/worker/cworker/c_worker.go:73]: Job incoming: {"name":"REPLICATION","id":"1fedc50249b5112da8af2e03","t":1581064239,"args":null}

Feb  7 08:32:54 172.18.0.1 jobservice[2587]: 2020-02-07T08:32:54Z [INFO] [/jobservice/api/server.go:101]: API server is stopped
Feb  7 08:32:54 172.18.0.1 jobservice[2587]: 2020-02-07T08:32:54Z [INFO] [/jobservice/hook/hook_agent.go:164]: Basic hook agent is stopped
Feb  7 08:32:54 172.18.0.1 jobservice[2587]: 2020-02-07T08:32:54Z [INFO] [/jobservice/period/enqueuer.go:89]: Periodic enqueuer is stopped
Feb  7 08:32:54 172.18.0.1 jobservice[2587]: 2020-02-07T08:32:54Z [INFO] [/jobservice/lcm/controller.go:110]: Status restoring loop is stopped
Feb  7 08:32:54 172.18.0.1 jobservice[2587]: 2020-02-07T08:32:54Z [INFO] [/jobservice/period/policy_store.go:115]: Periodical job policy store is stopped
Feb  7 08:32:54 172.18.0.1 jobservice[2587]: 2020-02-07T08:32:54Z [INFO] [/jobservice/period/basic_scheduler.go:58]: Basic scheduler is stopped

Feb  7 08:33:00 172.18.0.1 jobservice[2587]: 2020-02-07T08:33:00Z [INFO] [/jobservice/runner/redis.go:61]: |^_^| Job 'REPLICATION:1fedc50249b5112da8af2e03' exit with success

Feb  7 08:37:21 172.26.0.1 jobservice[2587]: 2020-02-07T08:37:21Z [INFO] [/jobservice/runtime/bootstrap.go:200]: API server is serving at 8080 with [http] mode at node [66fc7b33b973:172.27.0.8]
```

**Comment by steven-zou** [comment]
From the above log,

jobservice first start time is                                                     `2020-02-07T08:09:44Z`
replication job 1fedc50249b5112da8af2e03 start to run at `2020-02-07T08:30:40Z`
jobservice is stopped at                                                           `2020-02-07T08:32:54Z`
replication job 1fedc50249b5112da8af2e03 stio to run at   `2020-02-07T08:33:00Z`
jobservice second start time is                                                `2020-02-07T08:37:21Z`

As the js is stopped, the `Basic hook agent is stopped` is also exit. In the versions less than 1.10, the hook event publishing is using async way, the stop of the js & hook agent will close the event channel and then no hook events will be published.

After 1.10, it will be changed to use sync way to send hooks. The issue will be gone.

**Comment by steven-zou** [comment]
Fix has been introduced in release 1.10.1. For the older versions, we have got an agreement that we'll not do the duplicated work to fix it.

Close the issue.
